src/ResourceManager/Profile/Commands.Profile/Models/RMProfileClient.cs

```diff
@@ -117,86 +118,116 @@ public bool TryRemoveContext(IAzureContext context)
                  !account.IsPropertySet(AzureAccount.Property.CertificateThumbprint))
                 ? ShowDialog.Always : ShowDialog.Never;
 
-            // (tenant and subscription are present) OR
-            // (tenant is present and subscription is not provided)
-            if (!string.IsNullOrEmpty(tenantId))
+            if (skipValidation)
             {
-                var token = AcquireAccessToken(
-                    account, 
-                    environment, 
-                    tenantId, 
-                    password, 
-                    promptBehavior,
-                    promptAction);
-                if (TryGetTenantSubscription(
-                    token, 
-                    account, 
-                    environment, 
-                    tenantId, 
-                    subscriptionId, 
-                    subscriptionName, 
-                    out newSubscription, 
-                    out newTenant))
+                if ((string.IsNullOrEmpty(subscriptionId) && string.IsNullOrEmpty(subscriptionName)) || string.IsNullOrEmpty(tenantId))
+                {
+                    throw new PSInvalidOperationException("Must supply Subscription Id/Name and Tenant Id when using -SkipValidation");
+                }
+
+                if (!string.IsNullOrEmpty(subscriptionId))
                 {
-                    account.SetOrAppendProperty(AzureAccount.Property.Tenants, new[] { newTenant.Id.ToString() });
+                    newSubscription = new AzureSubscription
```

Comment:
We should set up the subscription with both ID and Name (if set)

Reply:
We will also want to fill out the additional properties in Subscription (TenantId, accountId?)

Reply:
Subscription is just one parameter with the possibility of having either name or ID, so there is no way for the user to give us both.

Reply:
Second part is done.